service/matching/handler/engine.go

```diff
@@ -149,15 +153,22 @@ func NewEngine(taskManager persistence.TaskManager,
 		partitioner:          partitioner,
 		timeSource:           timeSource,
 	}
+
+	go e.subscribeToMembershipChanges()
+
 	e.waitForQueryResultFn = e.waitForQueryResult
 	return e
 }
 
 func (e *matchingEngineImpl) Start() {
-	// As task lists are initialized lazily nothing is done on startup at this point.
+	// reset the shutdown channel if there's any listeners
+	if e.isShuttingDown() {
+		e.shutdown = make(chan struct{})
```

Comment:
this seems not quiet right. we shouldn't re-set this channel after initialization. Are you doing this to make the engine subsequently startable?

Reply:
yes, I'm trying to handle the case that stop may be called then subsequently start

Reply:
I'm a little unclear to what extent this is a problem, considering the original thought I had with this being affected by drains is no longer relevent. Removing.